py-macs2: add version 2.2.7.1 and support python@3.10:

glennpj opened the pull request:
The tarball from PyPi includes the Cythonized C files. The tarball from github does not. Remove the Cythonized C files from the source so that they are rebuilt with the Spack Python/Cython combination. This is necessary for python-3.10 but make sense for other combinations as well.

glennpj committed:
py-macs2: add version 2.2.7.1 and support python@3.10:

The tarball from PyPi includes the Cythonized C files. The tarball from
github does not. Remove the Cythonized C files from the source so that
they are rebuilt with the Spack Python/Cython combination. This is
necessary for python-3.10 but make sense for other combinations as well.

adamjstewart commented on var/spack/repos/builtin/packages/py-macs2/package.py:
```diff
@@ -13,13 +15,38 @@ class PyMacs2(PythonPackage):
+    patch(
+        "https://github.com/macs3-project/MACS/pull/497.patch?full_index=1",
+        sha256="eaff891b9b3c6a910bd5d454dcc6e21288c8d1ad4d6d6f77e370bc8f90921cbd",
+        when="^python@3.10:",
```
Does this patch work for all versions? I would at least add `@:2.2.7.1` since it won't be needed in the future (if there ever is a new release someday).

glennpj replied:
The 2.1 versions would not need it because they are python-2 only. I missed version 2.2.4 in there. Technically, it would need the patch, but I see that it does not apply cleanly to that version.

glennpj replied:
Ah, version 2.2.4 does not build with python-3.10 anyway so a version constraint is in order.

adamjstewart commented on var/spack/repos/builtin/packages/py-macs2/package.py:
```diff
@@ -13,13 +15,38 @@ class PyMacs2(PythonPackage):
+
+    depends_on("python@3.6:", when="@2.2.7.1:", type=("build", "run"))
+    depends_on("python@3.5:", when="@2.2:", type=("build", "run"))
+    depends_on("python@2.7:2.8", when="@:2.1", type=("build", "run"))
```
Spack no longer supports Python 2, you can remove all 2.1 versions

glennpj committed:
Edits based on review

- set python version constraint on version 2.2.4
- removed all python-2 versions and related constraints.
